### DISCUSSION TITLE
Can't remove tofus automatically when quit emacs with savehist on

### DISCUSSION THREAD

**Discussion by not-null-nil** [minad/consult#907] [Q&A] [answered]
Turning on consult and savehist same time, when quit emacs, it gives warning to select coding system, preventing quit.

Warning:
```
These default coding systems were tried to encode the following
problematic characters in the buffer ‘ session ’:
```

I look into the session, and it turns out to be caused by tofus from `consult--buffer-history` and `consult--line-history`.

I have looked some issues and know that candidate transformation function can be used to remove them, is there any config I miss to enable the function?

My config:
```
(use-package embark
  :config
  (setq prefix-help-command 'embark-prefix-help-command)
  :bind
  ("C-x ;" . #'embark-act))

(use-package consult
  :bind
  ("C-s" . #'consult-line)
  ("C-c i" . #'consult-imenu)
  ("C-x b" . #'consult-buffer)
  ("C-x C-r" . #'consult-recent-file)
  ("C-c k" . #' consult-ripgrep))

(use-package embark-consult
  :after (consult embark))

(use-package savehist
  :ensure nil
  :hook (after-init . savehist-mode)
  :init (setq enable-recursive-minibuffers t ; Allow commands in minibuffers
	      history-length 1000
	      savehist-additional-variables '(mark-ring
					      global-mark-ring
					      search-ring
					      regexp-search-ring
					      extended-command-history)
	      savehist-autosave-interval 300))
```
Tanks to your fantastic work.

**Comment by minad** [comment]
This query should not appear. Is there something special about your setup, encoding settings? Can you please create a minimal recipe based on emacs -Q? See https://github.com/minad/consult#bug-reports.

**Comment by not-null-nil** [comment, marked as answer]
After some test based on emasc -Q, it turns out to be caused by session

My steps:
1. Remove and reinstall consult and session
2. run emacs with emacs -Q
3. 
```
(package-initialize)
(require 'consult)
(require 'session)
``` 
5. use `M-x consult-buffer` to switch buffer
6. use `M-x session-save-session` to save session, and this is when the Warning show up.

It looks like the least version of session is 2021, and I think add an advice to `session-save-session` will make this work. I will give it a try.
